### DISCUSSION TITLE
Accessing different instances through

### DISCUSSION THREAD

**Discussion by sivag-csod** [jeremyevans/sequel#2218] [General]
My client wants to separate their data into separate instances. So our current code bases using Active record which we are unable to support. So we are thinking to move to sequel. So we tried that using following snippet and it worked. So my question is does sequel provide thread safety for each db instance if we are trying to access different DBs using raw queries instead models. My client data exists data both in mysql and postgres as well. 

```
DB1 = Sequel.connect(adapter: :postgres, user: 'postgres', password: 'postgres', host: 'localhost',
  port: 5000,
  database: 'postgres', max_connections: 10)
                                              
  DB2 = Sequel.connect(adapter: :postgres, user: 'postgres', password: 'postgres', host: 'localhost',
  port: 6000,
  database: 'postgres', max_connections: 10)

thread1 = Thread.new { pp  DB1[:books].all }
thread2 = Thread.new { pp DB2[:books].all }

[thread1, thread2].join()
```

is there better way to achieve this?

**Comment by janko** [comment]
Yes, each Sequel database has its own independent thread-safe connection pool, and its general thread-safety isn't compromised when there are multiple database instances.
